src/grammar.rs

```diff
@@ -307,6 +307,39 @@ impl<Pat> BitOr for RuleWithNamedFields<Pat> {
     }
 }
 
+#[derive(Copy, Clone, Debug, Ord, PartialOrd, Eq, PartialEq, Hash)]
+pub enum RepeatSep<T> {
+    Simple(T),
+    Trailing(T),
+    None,
```

Comment:
Hmm, I don't remember what we discussed, was I suggesting the `None` variant?
Either way, I'll take a closer look tomorrow.

Reply:
I had `RepeatMore`/`RepeatMoreTrailing` before, you suggested folding the `RepeatMoreTrailing` into `RepeatMore` (ditto for `Many`.

It's a three state option: `_+`, `_+ % _`, `_+ %% _`, thus me replacing `RepeatMore`'s `Option<Self>` with `RepeatSep<Self>`. Making a `Option<(Self, SepKind)>` could also work, but I thought the more semantic enum more appropriate.

Reply:
I think `Option<RepeatSep<T>>` would be fine?